seacatauth/external_login/authentication/providers/saml2.py

```
@@ -0,0 +1,164 @@
+
+	async def prepare_auth_request(self, state: dict, **kwargs) -> typing.Tuple[dict, aiohttp.web.Response]:
+		_, authn_request = self.SamlClient.prepare_for_authenticate(
+			binding=saml2.BINDING_HTTP_REDIRECT,
+			relay_state=state["state_id"],
+		)
+		assert authn_request["method"] == "GET"
+		auth_uri = dict(authn_request["headers"])["Location"]
+		return state, aiohttp.web.HTTPFound(auth_uri)
+
+
+	async def process_auth_callback(self, request: aiohttp.web.Request, payload: dict, state: dict, **kwargs) -> dict:
```

Ruff (0.13.2)

Unused method argument: `request`

(ARG002)